openfl/experimental/workflow/interface/fl_spec.py

```diff
@@ -132,7 +123,110 @@ def runtime(self, runtime) -> None:
             raise TypeError(f"{runtime} is not a valid OpenFL Runtime")
         self._runtime = runtime
 
-    def _capture_instance_snapshot(self, kwargs):
+    def run(self) -> None:
+        """Starts the execution of the flow."""
+        # Submit flow to Runtime
+        if str(self._runtime) == "LocalRuntime":
+            self._run_local()
+        elif str(self._runtime) == "FederatedRuntime":
+            self._run_federated()
+        else:
+            raise Exception("Runtime not implemented")
+
+    def _run_local(self) -> None:
+        """Executes the flow using LocalRuntime."""
+        self._setup_initial_state()
+        try:
+            # Execute all Participant (Aggregator & Collaborator) tasks and
+            # retrieve the final attributes
+            # start step is the first task & invoked on aggregator through
+            # runtime.execute_task
+            final_attributes = self.runtime.execute_task(
+                self,
+                self.start,
+            )
```

Comment:
As discussed on a previous call, it would be more elegant if the runtime were a separate object form the `FLSpec`. In this case we could write something like:
```python
final_attributes = runtime.execute_task(fl_spec, fl_spec.start)
```

Of course, this isn't a blocker for the current PR, and can be addressed later.

Reply:
This is a good suggestion and under discussion. Implementing this change would also require modifications to the LocalRuntime, as one of the key objectives of the Workflow Interface is to enable seamless migration between Simulation and Deployment modes for users. Considering this my proposal is to evaluate the impacts of this approach as part of a separate proposal for refactoring. Will follow-up on this as part of subsequent releases